### ISSUE TITLE
fetchBalance() in typescript only has the info feature defined in the Balances interface

### ISSUE THREAD

**Issue by dorienh** [ccxt/ccxt#18133]
### Operating System

iOS

### Programming Languages

JS

### CCXT Version

3.1.24

### Description

I am trying to get the balance of a specific asset using typescript. This works with js, but in ts the `total` feature is not defined

```
const balanceBefore = await exchange.fetchBalance(asset);
const assetBalanceBefore = balanceBefore.total[asset];
```

When I examine the object returned I only see a json with main `info` property. This will give me the info (although still figuring out how to get it out of there), but according to the [API](https://docs.ccxt.com/#/?id=balance-structure), it should return a Balance Structure with many different features. 

Looking at line 75 of `binance.d.ts`, I see that only info is promised: 

```
fetchBalance(params?: {}): Promise<{
        info: any;
    }>;
```

Trying this on Binance futures and Binance spot. 

### Code

```
  

```

**Comment by dorienh** [comment]
Maybe I went way too far here, but I finally got the availableAmount out. Seems like this kind of interface should be built in. 


[          const balanceBefore = await exchange.fetchBalance(asset);

          function findMyAsset(asset: AssetBalance, targetAsset: string) {
            return asset.asset === targetAsset;
          }

          const assetDataBefore = balanceBefore.info.assets.find(findMyAsset, asset) as AssetBalance;
          const availableBalance = assetDataBefore.availableBalance](url)



with these interfaces: 

```
  export interface AssetBalance {
    asset: string;
    walletBalance: string;
    unrealizedProfit: string;
    marginBalance: string;
    maintMargin: string;
    initialMargin: string;
    positionInitialMargin: string;
    openOrderInitialMargin: string;
    maxWithdrawAmount: string;
    crossWalletBalance: string;
    crossUnPnl: string;
    availableBalance: string;
    marginAvailable: boolean;
    updateTime: string;
  }

  export interface AccountData {
    feeTier: string;
    canTrade: boolean;
    canDeposit: boolean;
    canWithdraw: boolean;
    updateTime: string;
    multiAssetsMargin: boolean;
    totalInitialMargin: string;
    totalMaintMargin: string;
    totalWalletBalance: string;
    totalUnrealizedProfit: string;
    totalMarginBalance: string;
    totalPositionInitialMargin: string;
    totalOpenOrderInitialMargin: string;
    totalCrossWalletBalance: string;
    totalCrossUnPnl: string;
    availableBalance: string;
    maxWithdrawAmount: string;
    assets: AssetBalance[];
    positions: string;
  }
```

**Comment by carlosmiei** [comment]
Hello @dorienh,

`info` represents the `raw` response returned by the exchange so we don't type it, but you're right, `fetchBalance` should be returning an object of the type `Balances` and not `{  info: any; }`. I will try to fix it, thanks for letting us know.

**Comment by carlosmiei** [comment]
@dorienh This issue should be fixed now (as of 3.1.28), can you update to the latest version and try again?

**Comment by dorienh** [comment]
I have updated. It still gives me the following error: 

_Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'Balance'.
  No index signature with a parameter of type 'string' was found on type 'Balance'.ts(7053)_

If I try something like: 

```
const balance = await exchange.fetchBalance();
const total = balance.total['BTC']
```

Then for type definition it relays me to types.d.ts, which lists (line 151): 
```
export interface Balances extends Dictionary<Balance> {
    info: any;
}
```
